tests/bson/bug0894-001.phpt

```diff
@@ -0,0 +1,43 @@
+--TEST--
+PHPC-849: BSON get_properties handlers leak during gc_possible_root() checks
+--FILE--
+<?php
+
+$factoryMethods = [
+    function() { return new MongoDB\BSON\Binary('foo', MongoDB\BSON\Binary::TYPE_GENERIC); },
+    function() { return new MongoDB\BSON\Decimal128('3.14'); },
+    function() { return new MongoDB\BSON\Javascript('function foo() { return bar; }', ['bar' => 42]); },
+    function() { return new MongoDB\BSON\MaxKey; },
+    function() { return new MongoDB\BSON\MinKey; },
+    function() { return new MongoDB\BSON\ObjectID; },
+    function() { return new MongoDB\BSON\Regex('foo', 'i'); },
+    function() { return new MongoDB\BSON\Timestamp(1234, 5678); },
+    function() { return new MongoDB\BSON\UTCDateTime; },
+];
+
+foreach ($factoryMethods as $factoryMethod) {
+    $objects = [];
+
+    /* GC_ROOT_BUFFER_MAX_ENTRIES is defined as 10000 on PHP 5.x and 10001 on
+     * PHP 7.1, so use the larger value to always trigger gc_possible_root(). */
+    for ($i = 0; $i < 10001; $i++) {
+        $objects[] = $factoryMethod();
```

Comment:
You probably don't have to do this. You can call it once or twice, and then call [gc_collect_cycles()](http://php.net/manual/en/function.gc-collect-cycles.php). But not important enough to change.